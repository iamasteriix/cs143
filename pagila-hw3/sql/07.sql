/*
 * List all actors with Bacall Number 2;
 * That is, list all actors that have appeared in a film with an actor that has appeared in a film with 'RUSSELL BACALL',
 * but do not include actors that have Bacall Number < 2.
 */

-- select all actors will Bacall Number 1
WITH bacall_1 AS (
    SELECT actor_id
    FROM actor
    JOIN film_actor USING (actor_id)
    WHERE film_id IN (
        SELECT film_id
        FROM film_actor
        JOIN actor actor_table USING (actor_id)
        WHERE actor_table.first_name || ' ' || actor_table.last_name = 'RUSSELL BACALL'
    ) AND first_name || ' ' || last_name != 'RUSSELL BACALL'
    GROUP BY actor_id
)

-- select actors with Bacall Number 2
SELECT DISTINCT a1.first_name || ' ' || a1.last_name AS "Actor Name"
FROM actor a1
JOIN film_actor fa1 ON (a1.actor_id = fa1.actor_id)
JOIN film_actor fa2 ON (fa1.film_id = fa2.film_id)
JOIN actor a2 ON (fa2.actor_id = a2.actor_id)
WHERE a2.actor_id IN (SELECT * FROM bacall_1)
    AND a1.actor_id NOT IN (SELECT * FROM bacall_1)
    AND a1.first_name || ' ' || a1.last_name != 'RUSSELL BACALL'
ORDER BY "Actor Name";
