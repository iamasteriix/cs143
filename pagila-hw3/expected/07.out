    Actor Name     
-------------------
 ADAM HOPPER
 ALBERT JOHANSSON
 ALEC WAYNE
 AL GARLAND
 ANGELA HUDSON
 AUDREY OLIVIER
 BELA WALKEN
 BEN HARRIS
 BEN WILLIS
 BETTE NICHOLSON
 BOB FAWCETT
 BURT TEMPLE
 CAMERON WRAY
 CARMEN HUNT
 CARY MCCONAUGHEY
 CHARLIZE DENCH
 CHRIS DEPP
 CHRISTIAN GABLE
 CHRISTIAN NEESON
 CHRISTOPHER WEST
 CUBA BIRCH
 DAN TORN
 DARYL CRAWFORD
 DEBBIE AKROYD
 ED CHASE
 ED MANSFIELD
 ELVIS MARX
 EWAN GOODING
 FAY WOOD
 FRED COSTNER
 GARY PHOENIX
 GENE HOPKINS
 GENE WILLIS
 GEOFFREY HESTON
 GREG CHAPLIN
 GREGORY GOODING
 GROUCHO SINATRA
 GROUCHO WILLIAMS
 HARRISON BALE
 HENRY BERRY
 HUMPHREY WILLIS
 JADA RYDER
 JENNIFER DAVIS
 JESSICA BAILEY
 JODIE DEGENERES
 JOHN SUVARI
 JON CHASE
 JULIA BARRYMORE
 JULIA FAWCETT
 KENNETH HOFFMAN
 KENNETH PALTROW
 KENNETH PESCI
 KEVIN GARLAND
 KIM ALLEN
 KIRK JOVOVICH
 LAURA BRODY
 LUCILLE DEE
 MAE HOFFMAN
 MATTHEW CARREY
 MATTHEW JOHANSSON
 MATTHEW LEIGH
 MENA TEMPLE
 MERYL ALLEN
 MICHAEL BENING
 MICHAEL BOLGER
 MILLA KEITEL
 MILLA PECK
 NICK DEGENERES
 NICK WAHLBERG
 OPRAH KILMER
 PENELOPE CRONYN
 PENELOPE GUINESS
 RALPH CRUZ
 REESE KILMER
 REESE WEST
 RENEE BALL
 RENEE TRACY
 RICHARD PENN
 RITA REYNOLDS
 RUSSELL CLOSE
 SCARLETT DAMON
 SPENCER DEPP
 SUSAN DAVIS
 THORA TEMPLE
 VAL BOLGER
 WARREN NOLTE
 WHOOPI HURT
 WOODY HOFFMAN
(88 rows)

